hare/eligibility/oracle.go

```diff
@@ -446,28 +491,28 @@ func (o *Oracle) computeActiveSet(ctx context.Context, targetEpoch types.EpochID
 func (o *Oracle) computeActiveWeights(
 	targetEpoch types.EpochID,
 	activeSet []types.ATXID,
-) (map[types.NodeID]uint64, error) {
-	weightedActiveSet := make(map[types.NodeID]uint64)
+) (map[types.NodeID]identityWeight, error) {
+	identities := map[types.NodeID]identityWeight{}
```

Comment:
We already know the expected size of the map and can avoid reallocations:

```suggestion
	identities := make(map[types.NodeID]identityWeight, len(activeSet))
```